test-app/src/main/java/com/mapbox/navigation/examples/core/MapboxJunctionActivity.kt

```diff
@@ -0,0 +1,258 @@
+        mapboxReplayer.stop()
+        mapboxReplayer.clearEvents()
+        mapboxReplayer.pushRealLocation(this, 0.0)
+        val replayEvents = replayRouteMapper.mapDirectionsRouteGeometry(route)
+        mapboxReplayer.pushEvents(replayEvents)
+        mapboxReplayer.seekTo(replayEvents.first())
+        mapboxReplayer.play()
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = LayoutActivityJunctionBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        mapboxMap = binding.mapView.getMapboxMap()
+        locationComponent = binding.mapView.getLocationComponentPlugin().apply {
+            setLocationProvider(navigationLocationProvider)
+            enabled = true
+        }
+        mapCamera = getMapCamera()
+        init()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        binding.mapView.onStart()
+        if (::mapboxNavigation.isInitialized) {
+            mapboxNavigation.registerRoutesObserver(routesObserver)
+            mapboxNavigation.registerLocationObserver(locationObserver)
+            mapboxNavigation.registerRouteProgressObserver(replayProgressObserver)
+            mapboxNavigation.registerBannerInstructionsObserver(bannerInstructionsObserver)
+        }
+    }
+
+    override fun onStop() {
+        super.onStop()
+        if (::mapboxNavigation.isInitialized) {
+            mapboxNavigation.unregisterRoutesObserver(routesObserver)
+            mapboxNavigation.unregisterLocationObserver(locationObserver)
+            mapboxNavigation.unregisterRouteProgressObserver(replayProgressObserver)
+            mapboxNavigation.unregisterBannerInstructionsObserver(bannerInstructionsObserver)
+        }
+        binding.mapView.onStop()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        binding.mapView.onDestroy()
+        mapboxNavigation.onDestroy()
+    }
+
+    override fun onLowMemory() {
+        super.onLowMemory()
+        binding.mapView.onLowMemory()
+    }
+
+    override fun onMapLongClick(point: Point): Boolean {
+        vibrate()
+        ifNonNull(navigationLocationProvider.lastLocation) {
+            val or = Point.fromLngLat(139.7745686,35.677573)
+            val de = Point.fromLngLat(139.784915,35.680960)
+            findRoute(or, de)
+        }
+        return false
+    }
+
+    private fun init() {
+        initNavigation()
+        initStyle()
+        routeLine.initialize(binding.mapView, mapboxNavigation)
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun initNavigation() {
+        mapboxNavigation = MapboxNavigation(
+            NavigationOptions.Builder(this)
+                .accessToken(getMapboxAccessTokenFromResources())
+                .locationEngine(ReplayLocationEngine(mapboxReplayer))
+                .build()
+        )
+        junctionApi = MapboxJunctionApi(getMapboxRouteAccessToken(this))
```

Comment:
have we considered passing NavigationOptions to MapboxJunctionApi?

Advocating for a shared entity. That way we're not giving every feature the accessToken and applicationContext.

Reply:
We can, but from our perspective we have different access token to render Maps and different access token to render junction and signboard. The token you pass into `NavigationOptions` is the one you use to render Maps. So, if I pass NavigationOptions to the API, the token would not allow me to fetch junction/signboards as it is expecting a different one.

Also, there are quite a lot of things in `NavigationOptions` which I don't think makes sense to pass in this API as it might be confusing.
cc @kmadsen